indexing-service/src/main/java/io/druid/indexing/overlord/TaskLockbox.java

```diff
@@ -170,10 +173,11 @@ public int compare(Pair<Task, TaskLock> left, Pair<Task, TaskLock> right)
   public TaskLock lock(final Task task, final Interval interval) throws InterruptedException
   {
     giant.lock();
-
     try {
+      if(!activeTasks.contains(task.getId())){
+        throw new IllegalStateException(String.format("Unable to grant lock to inactive Task [%s]", task.getId()));
```

Comment:
yes, it does accounts for that, 
A task is marked as active when it gets added to the TaskQueue.